http/client.go

```diff
@@ -0,0 +1,412 @@
+	methodURL.RawQuery = url.Values{"version_id": []string{versionId}}.Encode()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, methodURL.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	var description client.CollectionDescription
+	if err := c.http.requestJson(req, &description); err != nil {
+		return nil, err
+	}
+	return &Collection{c.http, description}, nil
+}
+
+func (c *Client) GetAllCollections(ctx context.Context) ([]client.Collection, error) {
+	methodURL := c.http.baseURL.JoinPath("collections")
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, methodURL.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	var descriptions []client.CollectionDescription
+	if err := c.http.requestJson(req, &descriptions); err != nil {
+		return nil, err
+	}
+	collections := make([]client.Collection, len(descriptions))
+	for i, d := range descriptions {
+		collections[i] = &Collection{c.http, d}
+	}
+	return collections, nil
+}
+
+func (c *Client) GetAllIndexes(ctx context.Context) (map[client.CollectionName][]client.IndexDescription, error) {
+	methodURL := c.http.baseURL.JoinPath("indexes")
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, methodURL.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	var indexes map[client.CollectionName][]client.IndexDescription
+	if err := c.http.requestJson(req, &indexes); err != nil {
+		return nil, err
+	}
+	return indexes, nil
+}
+
+func (c *Client) ExecRequest(ctx context.Context, query string) *client.RequestResult {
+	methodURL := c.http.baseURL.JoinPath("graphql")
+	result := &client.RequestResult{}
+
+	body, err := json.Marshal(&GraphQLRequest{query})
+	if err != nil {
+		result.GQL.Errors = []error{err}
+		return result
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, methodURL.String(), bytes.NewBuffer(body))
+	if err != nil {
+		result.GQL.Errors = []error{err}
+		return result
+	}
+	c.http.setDefaultHeaders(req)
+
+	res, err := c.http.client.Do(req)
+	if err != nil {
+		result.GQL.Errors = []error{err}
+		return result
+	}
+	if res.Header.Get("Content-Type") == "text/event-stream" {
+		result.Pub = c.execRequestSubscription(ctx, res.Body)
+		return result
+	}
+	defer res.Body.Close() //nolint:errcheck
+
+	data, err := io.ReadAll(res.Body)
+	if err != nil {
+		result.GQL.Errors = []error{err}
+		return result
+	}
+	var response GraphQLResponse
+	if err = json.Unmarshal(data, &response); err != nil {
+		result.GQL.Errors = []error{err}
+		return result
+	}
+	result.GQL.Data = response.Data
+	result.GQL.Errors = response.Errors
+	return result
+}
+
+func (c *Client) execRequestSubscription(ctx context.Context, r io.ReadCloser) *events.Publisher[events.Update] {
+	pubCh := events.New[events.Update](0, 0)
+	pub, err := events.NewPublisher[events.Update](pubCh, 0)
+	if err != nil {
+		return nil
+	}
+
+	go func() {
+		eventReader := sse.NewReadCloser(r)
+		defer eventReader.Close() //nolint:errcheck
```

Comment:
todo: Same as other comment, please handle the error or document why not

Reply:
Added comment

Reply:
Please push it :)